internal/generator/edwards/template/tests/point.go.tmpl

```diff
@@ -800,4 +800,100 @@ func BenchmarkNeg(b *testing.B) {
 			point.Neg(&point)
 		}
 	})
+}
+
+func BenchmarkMixedAdd(b *testing.B) {
+	params := GetEdwardsCurve()
+	var s big.Int
+	s.SetString("52435875175126190479447705081859658376581184513", 10)
+	var point PointAffine
+	point.ScalarMultiplication(&params.Base, &s)
+
+	b.Run("Projective", func(b *testing.B) {
+		var accum PointProj
+		accum.setInfinity()
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			accum.MixedAdd(&accum , &point)
+		}
+	})
+	b.Run("Extended", func(b *testing.B) {
+		var accum PointExtended
+		accum.setInfinity()
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			accum.MixedAdd(&accum , &point)
+		}
+	})
+}
+
+func BenchmarkAdd(b *testing.B) {
+	params := GetEdwardsCurve()
+	var s big.Int
+	s.SetString("52435875175126190479447705081859658376581184513", 10)
+	
+	b.Run("Affine", func(b *testing.B) {
+		var point PointAffine
+		point.ScalarMultiplication(&params.Base, &s)
+		var accum PointAffine
+		accum.setInfinity()
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			accum.Add(&accum, &point)
+		}
+	})
+	b.Run("Projective", func(b *testing.B) {
+		var pointAff PointAffine
+		pointAff.ScalarMultiplication(&params.Base, &s)
+		var accum, point PointProj
+		point.FromAffine(&pointAff)
+		accum.setInfinity()
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			accum.Add(&accum, &point)
+		}
+	})
+	b.Run("Extended", func(b *testing.B) {
+		var pointAff PointAffine
+		pointAff.ScalarMultiplication(&params.Base, &s)
+		var accum, point PointExtended
+		point.FromAffine(&pointAff)
+		accum.setInfinity()
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			accum.Add(&accum, &point)
+		}
+	})
+}
+
+func BenchmarkIsOnCurve(b *testing.B) {
+	params := GetEdwardsCurve()
+	var s big.Int
+	s.SetString("52435875175126190479447705081859658376581184513", 10)
+
+	b.Run("positive", func(b *testing.B) {
+		var point PointAffine
+		point.ScalarMultiplication(&params.Base, &s)
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_ = point.IsOnCurve()
+		}
+	})
+
+	b.Run("negative", func(b *testing.B) {
+		var point PointAffine
+		point.ScalarMultiplication(&params.Base, &s)
+		point.X.Add(&point.X, &point.X)
```

Comment:
Forcing X to be out of the curve. I double-checked that `IsOnCurve` returns `false` in this case.

Reply:
hmm ok it works, if you checked with all Base points on all curves. Anyway, if `(x,y)` is on curve then `-a*x^2+y^2 == 1 + d*x^2*y^2` but `-a*(2x)^2+y^2 != 1 + d*(2x)^2*y^2` because none of the Base points have `x=0` or `y^2=-a/d` I guess.

Reply:
Good point. I could add a further check before the benchmark look that tests that on-call returns `false`, and fail fast if that isn't the case.

That way we could confirm that what's being benchmarked is actually correct.

Reply:
@yelhousni, pushed an extra commit with this if you want to take a look.

Reply:
yeah better, if end up adding more curve 👍